### PR TITLE
Use API gateway instead of directly calling Lambda functions

### PR DESCRIPTION
**Pull request by kmarcisz**
Replace the calls to AWS SDK directly to use API gateway and simpler calls utilizing `fetch` and `promise`.

The Lambda app has already been set up with API gateway access. To deploy this PR
- lambda needs to be updated with `lambda/app.js` code & deployed
- Easel app would need to be updated with the new `easel/app.js` code
  - `apiGateway` and `apiKey` variables can be found in AWS Lambda and are needed to deploy
  - existing dependencies can be removed
  - new dependencies (`clipper`, `easel`, `fetch` and `promise`) need to be added.

I have tested this using `TestSandbox` lambda and my own app that you can find in Easel apps called `DXF Import (testing)`.

### CODE REVIEW

**Comment by kmarcisz** [comment]
This PR is best reviewed with whiteline changes ignored.

**Comment by hphu3** [comment]
@kmarcisz can you link the doc used for the similar changes to the halftoner app?

**Comment by kmarcisz** [comment]
Document showing how to enable API Gateway for lambda function added at https://github.com/inventables/easel/wiki/Enabling-API-Gateway-for-Lambda-function

File: easel/app.js [hunk 1]
```diff
@@ -88,28 +87,34 @@ var executor = function (args, success, failure) {
...
+  })
+    .then(response => response.json())
+    .then(function(response) {
+      /**
+       * The lambda app finshed! result contains the apps payload, parsed as JSON
+       * The example app returns an svg as the only item in the map
+       *
+       * Since this is a "v1" app, we return SVG data
+       */
+      result = (typeof(response) == "string") ? JSON.parse(response) : response;
```
Comment (hphu3) [inline easel/app.js]: does `response.json()` not return a json object consistently?

File: easel/app.js [hunk 2]
```diff
@@ -88,28 +87,34 @@ var executor = function (args, success, failure) {
...
+
+      /**
+       * Process errors from inside of the Lambda app if returned payload has any
+       */
+      if ("errorMessage" in result) {
+        failure(result.errorMessage);
+      }
+
+      if ("svg" in result) {
```
Comment (hphu3) [inline easel/app.js]: do we want to warn or log something if there's no svg?

File: easel/app.js (hunk 1 above)
Comment (kmarcisz) [reply easel/app.js]: This was a leftover from the halftoner app change. Addressed in https://github.com/inventables/dxfimport/pull/3/commits/0acb0e9d0d3a3d9e0956f5b3461a43fdb4b77f0c

File: easel/app.js (hunk 2 above)
Comment (kmarcisz) [reply easel/app.js]: Addressed in https://github.com/inventables/dxfimport/pull/3/commits/2fd41a0fb46a4382557f683a55e9a09b551707de

**Pull request merged**